samples/nrf_desktop/src/hw_interface/buttons.c

```diff
@@ -335,15 +346,20 @@ void button_pressed(struct device *gpio_dev, struct gpio_callback *cb,
 static void init_fn(void)
 {
 	/* Setup GPIO configuration */
-	gpio_dev = device_get_binding(DT_GPIO_P0_DEV_NAME);
-	if (!gpio_dev) {
-		LOG_ERR("Cannot get GPIO device binding");
-		return;
+	for (size_t i = 0; i < ARRAY_SIZE(port_map); i++) {
+		gpio_devs[i] = device_get_binding(port_map[i]);
+		if (!gpio_devs[i]) {
+			LOG_ERR("Cannot get GPIO device binding");
+			return;
```

Comment:
please `goto error` where `module_set_state(MODULE_STATE_ERROR);` is called

Reply:
also for other errors

Reply:
done